server/routes/createGoal/routerRequestHandlers.ts

```diff
@@ -45,6 +45,21 @@ const checkAddStepFormsArrayExistsInSession = async (req: Request, res: Response
   }
 }
 
+/**
+ * Request handler function to check the AddNoteForm exists in the session for the prisoner reference in the
+ * request URL.
+ */
+const checkAddNoteFormExistsInSession = async (req: Request, res: Response, next: NextFunction) => {
+  if (!req.session.addNoteForm) {
+    logger.warn(
+      `No AddNoteForm object in session - user attempting to navigate to path ${req.path} out of sequence. Redirecting to start of Create Goal journey.`,
```

Comment:
`Redirecting to start of Create Goal journey.`
perhaps
`Redirecting to Add Note screen.` ?